### ISSUE TITLE
Elvis ?: [enhacement]

### ISSUE THREAD

**Issue by pablogrisafi1975** [amelentev/java-oo#28]
Thanks again for your extraordinary work.
You said that adding new operator is difficult because of no extension points at that stage. What about the elvis ?: operator, better known as null coalescing operator ? you know, instead of
<code>
     String firstName = user == null ? null : user.getFirstName() 
</code>
i'd like to write
<code>
     String firstName = user?:getFirstName() 
</code>
I don't know if that counts as a *new* operator, since both chars already have a meaning

**Comment by almson** [comment]
It doesn't parse, so I'm guessing it cannot be done. But how about a lady gaga operator `?$:`?

**Comment by pablogrisafi1975** [comment]
is that possible? At first I thought it was a joke, but then I saw it as

    String firstName = user ? $ : getFirstName() 

And I assume somehow $ is a magical variable that doesn't mess with the type system.

Mi ignorance about the whole source file -> class file process is almost perfect, so I don't know what can be done in each stage.

If it is possible, it is a nice workaround, and while somehow strange, it is way better than

    String firstName = user == null ? null : user.getFirstName()

**Comment by almson** [comment]
Yes, it should be possible if the $ variable is defined in the background
(the way lombok defines 'val'). However, your last example is the null-safe
dereference operator, which is different from elvis, and is usually `?.`.
Anyway, I don't know much about the limits of compiler plugins either.
